source/validate_id.cpp

```diff
@@ -417,7 +417,7 @@ bool idUsage::isValid<SpvOpTypeArray>(const spv_instruction_t* inst,
   } else if (5 == constInst->words.size()) {
     uint64_t value =
         constInst->words[3] | ((uint64_t)constInst->words[4]) << 32;
-    bool signedness = constResultType->second.inst->words[3];
+    bool signedness = constResultType->second.inst->words[3] ? true : false;
```

Comment:
I'd have not sent the pull if it were pointless :/

This occurs at warning level 3 in VC++, which is a common setting:
```
1>..\third_party\spirv-tools\source\validate_id.cpp(420): warning C4800: 'const unsigned int': forcing value to bool 'true' or 'false' (performance warning)
```
https://msdn.microsoft.com/en-us/library/b6801kcy.aspx

Irrespective of the reason the VC++ warning gives, implicitly casting a non-bool to a bool is not very readable. Were the value of the given word here (which is typed as uint32_t) be anything but 1 or 0 the behavior is hard to follow. Readability-wise, knowing the author intended to always treat any value not equal to zero as true is easier to parse from this (or related methods, such as !! commonly used in javascript or not not in python). Implicit casts from different width integer types is not great in modern C++ where bool has a very well-defined meaning and reliance on implicit or C-style casts is non-idiomatic.